apps/api/src/controllers/keys.controller.ts

```diff
@@ -1,17 +1,52 @@
-import { Controller, Get, HttpCode, HttpStatus, Logger, Param, HttpException } from '@nestjs/common';
-import { ApiOkResponse, ApiOperation, ApiTags } from '@nestjs/swagger';
+import { Controller, Get, HttpCode, HttpStatus, Logger, Param, HttpException, Post, Body } from '@nestjs/common';
+import { ApiBody, ApiOkResponse, ApiOperation, ApiTags } from '@nestjs/swagger';
 import { KeysService } from '../services/keys.service';
-import { KeysResponse } from '../../../../libs/common/src/types/dtos/keys.dto';
+import {
+  AddKeysRequest,
+  DeleteKeysRequest,
+  KeysResponse,
+  PublishKeysRequest,
+} from '../../../../libs/common/src/types/dtos/keys.dto';
+import { EnqueueService } from '../../../../libs/common/src/services/enqueue-request.service';
+import { TransactionType } from '../../../../libs/common/src';
 
 @Controller('keys')
 @ApiTags('keys')
 export class KeysController {
   private readonly logger: Logger;
 
-  constructor(private keysService: KeysService) {
+  constructor(
+    private keysService: KeysService,
+    private enqueueService: EnqueueService,
+  ) {
     this.logger = new Logger(this.constructor.name);
   }
 
+  @Post('add')
+  @HttpCode(HttpStatus.OK)
+  @ApiOperation({ summary: 'add new control keys for an MSA ID' })
+  @ApiOkResponse({ description: 'Found public keys.' })
+  @ApiBody({ type: AddKeysRequest })
+  /**
+   * Gets public keys.
+   * @param queryParams - The query parameters for getting the public keys.
+   * @returns A promise that resolves to an array of public keys associated with the given msaId.
+   * @throws An error if no public keys can be found.
+   */
+  async addKey(@Body() addKeysRequest: AddKeysRequest): Promise<string> {
+    try {
+      const { referenceId } = await this.enqueueService.enqueueRequest<PublishKeysRequest>({
+        ...addKeysRequest,
+        type: TransactionType.ADD_KEY,
+      });
+      this.logger.log(`AddKey in progress. referenceId: ${referenceId}`);
+      return `AddKey in progress. referenceId: ${referenceId}`;
```

Comment:
nit: I think we should think about returning JSON objects. I would like to compare to what the social-app-template is currently doing, it will depend on how the usage with the webhook shakes out.

Reply:
I'm into it. Happy to change.